nautobot/extras/api/views.py

```diff
@@ -508,7 +508,7 @@ def _run_job(request, job_model, legacy_response=False):
     if job_model.has_sensitive_variables:
```

Comment:
Can this fall through and end up creating a schedule if these conditions are met? 

- Approval is required
- Has sensitive variables 
- Scheduled for immediate execution

Falls through here:
- `job_model.has_sensitive_variables = True`
- `request.data["schedule"]["interval"] = "immediate"`

Falls through on L546:
- `schedule_data` is not None because `request.data["schedule"]` is set

Falls through on L549
- `job_model.approval_required = True`

`ScheduledJob` gets created in `_create_schedule()`, saving `job_kwargs` to the database.

Reply:
This appears to be a gap in the implementation from #2091.

Reply:
#2201 opened to track this separately